src/status_im2/common/contacts/events.cljs

```diff
@@ -0,0 +1,100 @@
+(ns status-im2.common.contacts.events
+  (:require [utils.re-frame :as rf]
+            [status-im.utils.types :as types]
+            [re-frame.core :as re-frame]
+            [status-im.ethereum.core :as ethereum]
+            [status-im.ethereum.ens :as ens]
+            [status-im.ethereum.stateofus :as stateofus]
+            [status-im.native-module.core :as status]
+            [status-im.utils.db :as utils.db]
+            [status-im.utils.utils :as utils]))
+
+(re-frame/reg-fx
+ :resolve-public-key-from-ens-name
+ (fn [{:keys [chain-id ens-name on-success on-error]}]
+   (ens/pubkey chain-id ens-name on-success on-error)))
+
+(re-frame/reg-fx
+ :decompress-public-key
+ (fn [{:keys [public-key on-success on-error]}]
+   (status/compressed-key->public-key
+    public-key
+    (fn [resp]
+      (let [{:keys [error]} (types/json->clj resp)]
+        (if error ; If response is of the form {:error "foo"} => ERROR
```

Comment:
The margin comments in this event handler seem largely unnecessary. The code speaks for itself IMO. Especially the one about the error.

Reply:
done